File: app/src/main/java/com/mqt/ganghuazhifu/utils/RealmUtils.java
package com.mqt.ganghuazhifu.utils;

import android.content.Context;

import io.realm.Realm;
import io.realm.RealmConfiguration;

/**
 * Created by msc on 2016/9/27.
 */

public class RealmUtils {

    private static RealmUtils mInstance;
    private String realName = "Message.Realm";

    private RealmUtils(Context context) {
        Realm.init(context);
        RealmConfiguration realmConfig = new RealmConfiguration.Builder().name(realName).build();
        Realm.setDefaultConfiguration(realmConfig);
    }

    public static RealmUtils getInstance(Context context) {
        if (mInstance == null) {
            synchronized (RealmUtils.class) {
                if (mInstance == null) {
                    mInstance = new RealmUtils(context);
                }
            }
        }
        return mInstance;
    }

    /**
     * 获得Realm对象
     *
     * @return
     */
    public Realm getRealm() {
        return Realm.getDefaultInstance();
    }

}

File: app/src/main/java/com/mqt/ganghuazhifu/utils/RealmUtils.kt
package com.mqt.ganghuazhifu.utils

import android.content.Context

import io.realm.Realm
import io.realm.RealmConfiguration

/**
 * Created by msc on 2016/9/27.
 */

class RealmUtils private constructor(context: Context) {
    private val realName = "Message.Realm"

    init {
        Realm.init(context)
        val realmConfig = RealmConfiguration.Builder().name(realName).build()
        Realm.setDefaultConfiguration(realmConfig)
    }

    /**
     * 获得Realm对象

     * @return
     */
    val realm: Realm
        get() = Realm.getDefaultInstance()

    companion object {

        private var mInstance: RealmUtils? = null

        fun getInstance(context: Context): RealmUtils? {
            if (mInstance == null) {
                synchronized(RealmUtils::class.java) {
                    if (mInstance == null) {
                        mInstance = RealmUtils(context)
                    }
                }
            }
            return mInstance
        }
    }

}
